docs/source/reference/index.rst

```diff
@@ -136,4 +136,13 @@ TensorFlow
    :toctree: generated/
    :nosignatures:
 
-   optuna_integration.TFKerasPruningCallback
+   optuna.integration.TFKerasPruningCallback
+
+XGBoost
+----------
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   optuna.integration.XGBoostPruningCallback
```

Comment:
```suggestion
   optuna_integration.XGBoostPruningCallback
```

Reply:
... and I will take care of this as well. Thank you!

Reply:
Additional corrections to index.rst and __init__.py done! ptal